richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/parser/markdown/MarkdownExport.kt

```diff
@@ -0,0 +1,108 @@
+package com.mohamedrejeb.richeditor.parser.markdown
+
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.style.TextDecoration
+import com.mohamedrejeb.richeditor.model.RichParagraph
+import com.mohamedrejeb.richeditor.model.RichSpan
+import com.mohamedrejeb.richeditor.model.RichSpanStyle
+import com.mohamedrejeb.richeditor.model.RichTextState
+
+private fun RichSpanStyle.encapsulateText(text: String): String {
+    return when (this) {
+        is RichSpanStyle.Code -> {
+            "`$text`"
+        }
+
+        is RichSpanStyle.Link -> {
+            "[$text](${this.url})"
+        }
+
+        is RichSpanStyle.Default -> text
+
+        else -> {
+            throw IllegalStateException("unknown rich span style")
+        }
+    }
+}
+
+internal fun RichSpan.toMarkdownText(): String {
+    if (this.isEmpty()) return ""
+    return if (this.children.isEmpty()) {
+        this.style.encapsulateText(
+            text = startMarkdown(this.spanStyle) + this.text + endMarkdown(this.spanStyle)
+        )
+    } else {
+        var markdownText = ""
+        markdownText += startMarkdown(this.spanStyle)
+        markdownText += this.text
+        for (child in this.children) {
+            markdownText += child.toMarkdownText()
+        }
+        markdownText += endMarkdown(this.spanStyle)
+        this.style.encapsulateText(markdownText)
+    }
+}
+
+internal fun RichParagraph.toMarkdownAsUnorderedList(): String {
+    var markdownText = ""
+    for (child in this.children) {
```

Comment:
Oh, I didn't know that the whole paragraph is an unordered list item

Reply:
The paragraph is representing any block element that requires a new line and a list item requires a line, even in the future if I'm going to add images support for example it's going to be treated as a paragraph since it's going to take an entire line by itself.